lib/task/import/validate/csvDigitalObjectPathValidator.class.php

```diff
@@ -0,0 +1,232 @@
+                }
+            } else {
+                if (empty($this->digitalObjectUses)) {
+                    $this->testData->addResult(sprintf("Column 'digitalObjectPath' is empty."));
+                } else {
+                    // Check for Paths that will be overridden by URI.
+                    if (0 < $this->overriddenByUriCount) {
+                        $this->testData->setStatusWarn();
+                        $this->testData->addResult(sprintf("'digitalObjectPath' will be overridden by 'digitalObjectUri' if both are populated."));
+                        $this->testData->addResult(sprintf("'digitalObjectPath' values that will be overridden by digitalObjectUri: %s", $this->overriddenByUriCount));
+                    }
+
+                    $digitalObjectPathsUsedMoreThanOnce = $this->getUsedMoreThanOnce();
+
+                    if (!empty($digitalObjectPathsUsedMoreThanOnce)) {
+                        $this->testData->setStatusWarn();
+                        $this->testData->addResult(sprintf('Number of duplicated digital object paths found in CSV: %s', count($digitalObjectPathsUsedMoreThanOnce)));
+
+                        foreach ($digitalObjectPathsUsedMoreThanOnce as $path) {
+                            $this->testData->addDetail(sprintf("Number of duplicates for path '%s': %s", $path, $this->digitalObjectUses[$path]));
+                        }
+                    }
+
+                    $unusedFiles = $this->getUnusedFiles();
+
+                    if (!empty($unusedFiles)) {
+                        $this->testData->setStatusWarn();
+                        $this->testData->addResult(sprintf('Digital objects in folder not referenced by CSV: %s', count($unusedFiles)));
+
+                        foreach ($unusedFiles as $file) {
+                            $this->testData->addDetail(sprintf('Unreferenced digital object: %s', $file));
+                        }
+                    }
+
+                    $missingFiles = $this->getMissingDigitalObjects();
+
+                    if (!empty($missingFiles)) {
+                        $this->testData->setStatusError();
+                        $this->testData->addResult(sprintf('Digital object referenced by CSV not found in folder: %s', count($missingFiles)));
+
+                        foreach ($missingFiles as $file) {
+                            $this->testData->addDetail(sprintf('Unable to locate digital object: %s/%s', $this->pathToDigitalObjects, $file));
+                        }
+                    }
+                }
+            }
+        }
+
+        return parent::getTestResult();
+    }
+
+    public function setPathToDigitalObjects(string $path)
+    {
+        $path = (!empty($path) && is_dir($path)) ? $path : '';
+
+        if (is_dir($path)) {
+            $this->pathToDigitalObjects = realpath($path) ? realpath($path) : $path;
```

Comment:
Cool, I didn't know about `realpath()` :)

Reply:
I think if `realpath($path)` returns false, it should be considered an error, instead of continuing with `$path`

Reply:
Returning $path in the case where realpath() fails is there to support the use of the VFS in the PHP Unit tests as realpath() always returns false.